core/includes/form.inc

```diff
@@ -3023,7 +3023,7 @@ function form_process_password_confirm($element) {
     '#title' => t('New password'),
     '#value' => empty($element['#value']) ? NULL : $element['#value']['pass1'],
     '#required' => $element['#required'],
-    '#attributes' => array('class' => array('password-field')),
+    '#attributes' => $element['#attributes'] + array('class' => array('password-field')),
```

Comment:
A quick question, didn't actually test it yet: doesn't that array merge require the `#attributes` key is already set and an array?

The definition in system_element_info() doesn't define it, though. So that could lead to nagging.

Reply:
Hi @indigoxela - thanks for checking one this. The `#attributes` key is always initialized for form elements before processing; this is done in `form_builder()`. See [this line](https://github.com/backdrop/backdrop/blob/91dd4955aec444f15db19a3ad9a6cbbbf204c08e/core/includes/form.inc#L1861).

Reply:
That's right, many thanks for checking. :+1: